3A-small_n_differential_expression.R

```diff
@@ -112,102 +111,147 @@ message(paste("Smaller subtype has", smaller_subtype_size, "samples,",
 # initialize list to hold Jaccard, Rand, Spearman data from the 10 trials
 stats.df.list <- list()
 
-# we're going to repeat the small n experiment 10 times
-for (trial.iter in 1:10) {
-
-  # for each n (3...50), get the sample names that will be included in the
-  # experiment and on each platform
-  sample.list <-
-    lapply(no.samples,  # for each n (3...50)
-           function(x) GetSamplesforMixingSmallN(x, sample.df,
-                                                 subtype = last(two_subtypes)))
-
-  # initialize list to hold differential expression results (eBayes output)
-  master.deg.list <- list()
-
-  for (smpl.no.iter in seq_along(sample.list)) {  # for each n (3...50)
-    # normalize data
-    norm.list <- SmallNNormWrapper(array.dt = array.dt,
-                                   seq.dt = seq.dt,
-                                   mix.list = sample.list[[smpl.no.iter]],
-                                   zto = FALSE)
-    # perform differential expression analysis
-    master.deg.list[[as.character(no.samples[smpl.no.iter])]] <-
-      SmallNDEGWrapper(norm.list = norm.list, sample.df = sample.df,
-                       subtype = last(two_subtypes))
+# Do this at 0-100% RNA-seq titration levels
+# parallel backend
+cl <- parallel::makeCluster(detectCores() - 1)
+doParallel::registerDoParallel(cl)
+
+# at each titration level (0-100% RNA-seq)
+stats.df.list[1:9] <- foreach(seq_prop = seq(0.1, .9, 0.1), .packages = c("tidyverse")) %dopar% {
```

Comment:
Assuming the indexing needs to be here because of how `foreach()` is working here?

Reply:
Yes, index is intended to correspond to each iteration of `foreach()`